web/react/components/sidebar.jsx

```diff
@@ -347,38 +356,63 @@ module.exports = React.createClass({
             }
 
             // set up click handler to switch channels (or create a new channel for non-existant ones)
-            var clickHandler = null;
-            var href;
+            var handleClick = null;
+            var href = '#';
             if (!channel.fake) {
-                clickHandler = function(e) {
+                handleClick = function clickHandler(e) {
                     e.preventDefault();
                     utils.switchChannel(channel);
                 };
-                href = '#';
             } else {
-                href = TeamStore.getCurrentTeamUrl() + '/channels/' + channel.name;
+                // It's a direct message channel that doesn't exist yet so let's create it now
+                var ids = channel.name.split('__');
```

Comment:
Yup, I'll make a function to handle getting the other user id from a PM channel name